autopush-common/src/endpoint.rs

```diff
@@ -0,0 +1,37 @@
+use crate::errors::{Result, ResultExt};
+use fernet::MultiFernet;
+use openssl::hash;
+use url::Url;
+use uuid::Uuid;
+
+/// Create an v1 or v2 WebPush endpoint from the identifiers
+///
+/// Both endpoints use bytes instead of hex to reduce ID length.
+//  v1 is the uaid + chid
+//  v2 is the uaid + chid + sha256(key).bytes
+pub fn make_endpoint(
+    uaid: &Uuid,
+    chid: &Uuid,
```

Comment:
As noted in https://github.com/mozilla-services/autopush-rs/pull/189, we want to make sure that we don't accidentally modify the channelid that's provided to us from the Client, since they treat them as strings. We may want to alter this to do the same.

Reply:
We need to have it as a `Uuid` here because we use the raw UUID bytes to compute the endpoint URL. Having it in hyphenated form (if that's what the client gave us) would produce an invalid URL.
(side note: we could just call `Uuid::as_bytes` instead of formatting as a simple UUID and then decoding the hex...)

Reply:
Yeah, we decode because it's far more efficient. At least we have an answer from client that they want lowercase, dashed so we can default to that when we compose messages out.